src/rdkafka_cgrp.c

```diff
@@ -968,6 +979,25 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                  * a matching assignor so error out early. */
                 ErrorCode = RD_KAFKA_RESP_ERR__BAD_MSG;
         }
+	else if (!ErrorCode) {
+                char *protocol_name = RD_KAFKAP_STR_DUP(&Protocol);
```

Comment:
Since this is only used in this stack frame use RD_KAFKAP_STR_DUPA and drop the free below.

Reply:
cool. the allocation did seem unfortunate.